dsc/workflows/base/__init__.py

```diff
@@ -19,10 +19,29 @@
 if TYPE_CHECKING:  # pragma: no cover
     from collections.abc import Iterator
 
+    from dsc.report import Report
+
 logger = logging.getLogger(__name__)
 CONFIG = Config()
 
 
+@dataclass
+class WorkflowEvents:
+    """Record of events during the execution of Workflow methods.
+
+    This dataclass is designed to hold useful data used in reporting.
+    It is comprised of three lists, which contain details
+    about reconciled, submitted, and processed items -- aligning with
+    the DSC CLI commands (reconcile, submit, and finalize). Error
+    messages are also tracked in a list.
+    """
+
+    reconciled_items: list[str] = field(default_factory=list)
+    submitted_items: list[str] = field(default_factory=list)
+    processed_items: list[dict] = field(default_factory=list)
```

Comment:
`finalize` vs `processed` injects some ambiguity and I'm not sure which is preferable, but DataEng should probably decide on that for consistency to prevent confusion

Reply:
Ah, the age-old question! 😅 

My interpretation:
> The `finalize` CLI command performs two functions: **processing** the DSS result messages sent to the output queue for a given workflow and sending the report (an email with attachments) to designated recipients. For the report, we are interested in knowing what DSS result messages sent to the output queue were _successfully_ **processed** (i.e., a valid DSS result message means it was successfully submitted to DSpace).

With that understanding, I think `processed_items` is a reasonable name for this collected data, but I am flexible!